internal/integration/api/trustedboot.go

```diff
@@ -0,0 +1,99 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+//go:build integration_api
+
+package api
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/safe"
+
+	"github.com/siderolabs/talos/internal/integration/base"
+	"github.com/siderolabs/talos/pkg/machinery/client"
+	runtimeres "github.com/siderolabs/talos/pkg/machinery/resources/runtime"
+)
+
+// TrustedBootSuite verifies Talos is securebooted.
+type TrustedBootSuite struct {
+	base.K8sSuite
+
+	ctx       context.Context //nolint:containedctx
+	ctxCancel context.CancelFunc
+}
+
+// SuiteName ...
+func (suite *TrustedBootSuite) SuiteName() string {
+	return "api.TrustedBootSuite"
+}
+
+// SetupTest ...
+func (suite *TrustedBootSuite) SetupTest() {
+	// make sure API calls have timeout
+	suite.ctx, suite.ctxCancel = context.WithTimeout(context.Background(), 1*time.Minute)
+}
+
+// TearDownTest ...
+func (suite *TrustedBootSuite) TearDownTest() {
+	if suite.ctxCancel != nil {
+		suite.ctxCancel()
+	}
+}
+
+// TestTrustedBootState verifies that the system is booted in secure boot mode
+// and that the disks are encrypted.
+func (suite *TrustedBootSuite) TestTrustedBootState() {
+	if testing.Short() {
+		suite.T().Skip("skipping in short mode")
+	}
+
+	if !suite.TrustedBoot {
+		suite.T().Skip("skipping as talos is not explicitly trusted booted")
+	}
+
+	node := suite.RandomDiscoveredNodeInternalIP()
+	ctx := client.WithNode(suite.ctx, node)
+
+	securityResource, err := safe.StateGetByID[*runtimeres.SecurityState](ctx, suite.Client.COSI, runtimeres.SecurityStateID)
```

Comment:
in all these cases, better to use `safe.StateWatchFor` with types `Created, Updated`, as this doesn't depend on the resource being there at the moment of the test run